back/src/Guide/Database/Set.hs

```diff
@@ -0,0 +1,584 @@
+import Guide.Utils (Uid (..), Url, moveDown, moveUp)
+
+
+----------------------------------------------------------------------------
+-- Categories
+----------------------------------------------------------------------------
+
+-- | Set category title.
+setCategoryTitle :: Uid Category -> "title" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryTitle catId (arg #title -> title) = do
+  let sql = [r|
+        UPDATE categories
+        SET title = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, title) (Statement sql encoder decoder False)
+
+-- | Set category group.
+setCategoryGroup :: Uid Category -> "group_" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryGroup catId (arg #group_ -> group_) = do
+  let sql = [r|
+        UPDATE categories
+        SET group_ = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, group_) (Statement sql encoder decoder False)
+
+-- | Set category notes.
+setCategoryNotes :: Uid Category -> "notes" :! Text -> ExceptT DatabaseError Transaction ()
+setCategoryNotes catId (arg #notes -> notes) = do
+  let sql = [r|
+        UPDATE categories
+        SET notes = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, notes) (Statement sql encoder decoder False)
+
+-- | Set new category notes.
+setCategoryStatus :: Uid Category -> CategoryStatus -> ExceptT DatabaseError Transaction ()
+setCategoryStatus catId status = do
+  let sql = [r|
+        UPDATE categories
+        SET status_ = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam categoryStatusParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, status) (Statement sql encoder decoder False)
+
+-- | Set category selections.
+setCategoryEnabledSections
+  :: Uid Category
+  -> Set ItemSection     -- ^ Sections to enable
+  -> Set ItemSection     -- ^ Sections to disable
+  -> ExceptT DatabaseError Transaction ()
+setCategoryEnabledSections catId toEnable toDisable = do
+  oldSections <- _categoryEnabledSections <$> getCategory catId
+  let newSections = (oldSections <> toEnable) Set.\\ toDisable
+  let sql = [r|
+        UPDATE categories
+        SET enabled_sections = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam itemSectionSetParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, newSections) (Statement sql encoder decoder False)
+
+-- | Add itemId to category items_order.
+addItemIdToCategory :: Uid Category -> Uid Item -> ExceptT DatabaseError Transaction ()
+addItemIdToCategory catId itemId = do
+    itemsOrder <- getCategoryItemsOrder catId
+    let addItemId =
+          if notElem itemId itemsOrder then itemId : itemsOrder else itemsOrder
```

Comment:
If the item was already in `items_order`, it's always a bug. It should be reported and not thrown away.